pkg/scheduler/factory/factory_test.go

```diff
@@ -616,3 +621,189 @@ func testGetBinderFunc(expectedBinderType, podName string, extenders []algorithm
 		t.Errorf("Expected binder %q but got %q", expectedBinderType, binderType)
 	}
 }
+
+type TestPlugin struct {
+	name string
+}
+
+func (t *TestPlugin) Name() string {
+	return t.name
+}
+
+func (t *TestPlugin) Score(pc *framework.PluginContext, p *v1.Pod, nodeName string) (int, *framework.Status) {
+	return 1, nil
+}
+
+func (t *TestPlugin) NormalizeScore(pc *framework.PluginContext, pod *v1.Pod, scores framework.NodeScoreList) *framework.Status {
+	return nil
+}
+
+func (t *TestPlugin) Filter(pc *framework.PluginContext, pod *v1.Pod, nodeInfo *schedulernodeinfo.NodeInfo) *framework.Status {
+	return nil
+}
+
+// Test configures a scheduler from a policies defined in a file
+// It combines some configurable predicate/priorities with some pre-defined ones
+func TestCreateWithFrameworkPlugins(t *testing.T) {
+	var configData []byte
+	var policy schedulerapi.Policy
+
+	client := fake.NewSimpleClientset()
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	predicateOneName := "PredicateOne"
+	filterOneName := "FilterOne"
+	predicateTwoName := "PredicateTwo"
+	filterTwoName := "FilterTwo"
+	predicateThreeName := "PredicateThree"
+	predicateFourName := "PredicateFour"
+	priorityOneName := "PriorityOne"
+	scoreOneName := "ScoreOne"
+	priorityTwoName := "PriorityTwo"
+	scoreTwoName := "ScoreTwo"
+	priorityThreeName := "PriorityThree"
+
+	configProducerRegistry := &frameworkplugins.ConfigProducerRegistry{
+		PredicateToConfigProducer: make(map[string]frameworkplugins.ConfigProducer),
+		PriorityToConfigProducer:  make(map[string]frameworkplugins.ConfigProducer),
+	}
+	configProducerRegistry.RegisterPredicate(predicateOneName,
+		func(_ frameworkplugins.ConfigProducerArgs) (config.Plugins, []config.PluginConfig) {
```

Comment:
nit: it seems that the you can define a generic producer func and reuse it for predicate one, two, etc.